### DISCUSSION TITLE
controller crashed wrong!

### DISCUSSION THREAD

**Discussion by xjs-xrhm** [cyberbotics/webots#4116] [Q&A]
My control program was able to run successfully before, but now it suddenly shows an error and can't run.
The error is as shown below

![image](https://user-images.githubusercontent.com/55490426/149441953-424ad980-fd8b-4bfc-b4e4-7477ab6c560d.png)

**Comment by ad-daniel** [comment]
What is the controllor? Can you share it?

**Reply by xjs-xrhm** [reply]
My main control is a quadruped robot, using a polynomial planning curve.
![image](https://user-images.githubusercontent.com/55490426/149461291-6d9e9c0d-b89b-436e-b549-7c85704a1841.png)
![image](https://user-images.githubusercontent.com/55490426/149461343-bdf9e554-fb12-495c-9584-5432bde003bf.png)

**Reply by xjs-xrhm** [reply]
I tried to use it and found that the programs related to inverse kinematics, I don't know if the amount of calculation is too large, which makes them unable to run.
Control programs such as those using a central pattern generator-based (CPG) approach can be run.
But I think even the calculation of inverse kinematics, the computer should be able to support it.

**Reply by ad-daniel** [reply]
Hard to say without the entire file, but the amount of computation on its own normally doesn't make the controller crash. The issue is often trying to access NULL pointers, or similar. To debug your controller a good way is to progressively comment out parts of the code until at some point the controller no longer crashes. When you know which is the part that makes it crash, finding the error is easier.

**Reply by xjs-xrhm** [reply]
My problem is solved!
The reason is that the output path of a file in my program is not written correctly. After changing the correct path, the control program runs normally.
